src/main/java/games/strategy/engine/lobby/server/login/LobbyLoginValidator.java

```diff
@@ -50,7 +50,7 @@
   public Map<String, String> getChallengeProperties(final String userName, final SocketAddress remoteAddress) {
     // we need to give the user the salt key for the username
     final Map<String, String> challenge = new HashMap<>();
-    final HashedPassword password = new UserController().getPassword(userName);
+    final HashedPassword password = new UserController().getMd5Password(userName);
```

Comment:
I'll try to explain it:
The "Problem" with hashed passwords is that they are well hashed and you can't simply convert them to another hashing algorithm.
Therefore we orinally needed to have both, the "original" MD5 hashed password and, once the users log back in, the newly hashed password, in order to migrate them one after another.
Once all passwords are migrated, we can safely release a breaking change, which only authentificates with the bcrypted password, as this authentification is secure from an cryptographic standpoint.
Until then we want to provide both ways, so users can login with any client they want.
The password hashed were not used for their designed behaviour, but rather as some sort of challenge-response system, where the hash replaces the actual password and therefore a DB leak would allow an attacker to login as anyone,